reddit_liveupdate/controllers.py

```diff
@@ -1090,3 +1092,38 @@ def GET_mediaembed(self, liveupdate, embed_index):
         }
 
         return pages.LiveUpdateMediaEmbedBody(**args).render()
+
+
+@add_controller
+class LiveUpdateAdminController(RedditController):
+    @validate(VAdmin())
+    def GET_happening_now(self):
+        candidate_thread_ids = queries.get_active_events()
+        candidate_threads = LiveUpdateEvent._byID(list(candidate_thread_ids),
+                return_dict=False)
+
+        candidate_threads = filter(lambda t: not (t.banned or
+                                                  t.nsfw or
+                                                  t.state == 'complete'),
+                                   candidate_threads)
+        current_thread_id = NamedGlobals.get('live_happening_now', None)
+        if current_thread_id:
+            current_thread = LiveUpdateEvent._byID(current_thread_id)
+        else:
+            current_thread = None
+        return AdminPage(content=pages.HappeningNowAdmin(candidate_threads, current_thread),
+                         title='live: happening now',
+                         nav_menus=[]).render()
+
+    @validate(
+        VAdmin(),
+        VModhash(),
+        featured_thread=VLiveUpdateEventUrl('url'),
+    )
+    def POST_happening_now(self, featured_thread):
+        if featured_thread:
+            featured_thread = featured_thread._id
```

Comment:
nit: it's usually cleaner not to rewrite a variable but rather to use a new name.